File: src/main/java/intricateengineers/intricatemachinery/core/ClientProxy.java
/*
 * Copyright (c) 2016 IntricateEngineers
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package intricateengineers.intricatemachinery.core;

import intricateengineers.intricatemachinery.client.event.DebugRenderHandler;
import intricateengineers.intricatemachinery.common.module.DummyModule;
import intricateengineers.intricatemachinery.common.module.FurnaceModule;
import net.minecraft.client.renderer.block.model.ModelResourceLocation;
import net.minecraftforge.client.event.ModelBakeEvent;
import net.minecraftforge.client.event.TextureStitchEvent;
import net.minecraftforge.common.MinecraftForge;
import net.minecraftforge.fml.common.event.FMLInitializationEvent;
import net.minecraftforge.fml.common.event.FMLPostInitializationEvent;
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;

/**
 * @author topisani
 */
@SuppressWarnings("unused")
public class ClientProxy extends CommonProxy {


    @SubscribeEvent
    public void onPostBake(ModelBakeEvent event) {
        event.getModelRegistry().putObject(new ModelResourceLocation(ModInfo.MOD_ID + ":furnace", "inventory"), FurnaceModule.MODEL.getBakedModel());
        event.getModelRegistry().putObject(new ModelResourceLocation(ModInfo.MOD_ID + ":furnace", "multipart"), FurnaceModule.MODEL.getBakedModel());
        event.getModelRegistry().putObject(new ModelResourceLocation(ModInfo.MOD_ID + ":dummy", "inventory"), DummyModule.MODEL.getBakedModel());
        event.getModelRegistry().putObject(new ModelResourceLocation(ModInfo.MOD_ID + ":dummy", "multipart"), DummyModule.MODEL.getBakedModel());
    }

    @SubscribeEvent
    public void onTextureStitch(TextureStitchEvent.Pre event) {
        FurnaceModule.MODEL.getBakedModel().initTextures();
        FurnaceModule.MODEL.init();
        DummyModule.MODEL.getBakedModel().initTextures();
        DummyModule.MODEL.init();
    }

    @SubscribeEvent
    public void onTextureStitch(TextureStitchEvent.Post event) {
        FurnaceModule.MODEL.getBakedModel().initQuads();
        DummyModule.MODEL.getBakedModel().initQuads();
    }

    @Override
    public void preInit(FMLPreInitializationEvent event) {
    }

    @Override
    public void init(FMLInitializationEvent event) {
        MinecraftForge.EVENT_BUS.register(DebugRenderHandler.instance);
    }

    @Override
    public void postInit(FMLPostInitializationEvent event) {
    }
}

File: src/main/scala/intricateengineers/intricatemachinery/core/ClientProxy.scala
package intricateengineers.intricatemachinery.core

import javax.annotation.Nullable

import intricateengineers.intricatemachinery.api.client.FrameRenderer
import intricateengineers.intricatemachinery.api.module.{FrameModel, MachineryFrame}
import intricateengineers.intricatemachinery.client.event.DebugRenderHandler
import intricateengineers.intricatemachinery.common.module.{DummyModel, FurnaceModel}
import mcmultipart.client.multipart.MultipartRegistryClient
import net.minecraft.block.state.IBlockState
import net.minecraft.client.renderer.block.model._
import net.minecraft.client.renderer.texture.TextureAtlasSprite
import net.minecraft.util.EnumFacing
import net.minecraftforge.client.event.{ModelBakeEvent, TextureStitchEvent}
import net.minecraftforge.common.MinecraftForge
import net.minecraftforge.fml.common.event.{FMLInitializationEvent, FMLPostInitializationEvent,
FMLPreInitializationEvent}
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent

object ClientProxy {
  var EMPTY_MODEL: IBakedModel = new IBakedModel() {
    def getQuads(@Nullable state: IBlockState, @Nullable side: EnumFacing, rand: Long): java.util.List[BakedQuad] = {
      new java.util.ArrayList[BakedQuad]
    }

    def isAmbientOcclusion: Boolean = false

    def isGui3d: Boolean = false

    def isBuiltInRenderer: Boolean = false

    def getParticleTexture: TextureAtlasSprite = null

    def getItemCameraTransforms: ItemCameraTransforms = null

    def getOverrides: ItemOverrideList = null
  }
}

class ClientProxy extends CommonProxy {
  @SubscribeEvent
  def onPostBake(event: ModelBakeEvent) {
    event.getModelRegistry.putObject(new ModelResourceLocation(MachineryFrame.NAME, "inventory"), ClientProxy
      .EMPTY_MODEL)
    event.getModelRegistry.putObject(new ModelResourceLocation(MachineryFrame.NAME, "multipart"), ClientProxy
      .EMPTY_MODEL)
  }

  @SubscribeEvent
  def onTextureStitch(event: TextureStitchEvent.Pre) {
    FurnaceModel.boxes.foreach(_.initTextures())
    DummyModel.boxes.foreach(_.initTextures())
    FrameModel.bakedModel.initTextures()
  }

  @SubscribeEvent
  def onTextureStitch(event: TextureStitchEvent.Post) {
    FurnaceModel.boxes.foreach(_.updateQuads())
    DummyModel.boxes.foreach(_.updateQuads())
    FrameModel.boxes.foreach(_.updateQuads())
  }

  override def preInit(event: FMLPreInitializationEvent) {
  }

  override def init(event: FMLInitializationEvent) {
    MinecraftForge.EVENT_BUS.register(DebugRenderHandler.instance)
    MultipartRegistryClient.bindMultipartSpecialRenderer(classOf[MachineryFrame], new FrameRenderer)
  }

  override def postInit(event: FMLPostInitializationEvent) {
  }
}
